### DISCUSSION TITLE
Form: adding relation on a one-to-many

### DISCUSSION THREAD

**Discussion by gvlx** [Budibase/budibase#3479] [Help]
I have a form on table "main" to a one-to-many relationship to another table "related".

I have a table showing the values of the "related" tables so the user can choose one of them.

I want the user to click on an icon of the chosen row of the "related" and have it added to the relationship of the "main" table.

Which options should I put on the `save form` action so that this new relation is added, while keeping the existing.

If the user wants to _remove that one row from the relation_, which options should I chose then?

Thanks

**Comment by mjashanks** [comment]
Hi @gvlx 

I have played around with this. I don't think it's possible right now. We would need this issue: https://github.com/Budibase/budibase/issues/3151, which should be released in a few weeks.

At that point, it should be possible - we will need to use Javascript to do it. By
1. Taking the existing array of related record
2. Pushing the new item to it
3. Saving the result to `State`
4. On Save, update the `main`'s relations to the value stored in state

Update State:

![image](https://user-images.githubusercontent.com/3524181/143072132-114645a7-e0e2-445f-9669-d007e07e1608.png)

The javascript inside this will look something like:
```javascript
const children = $("Repeater.Parent.Child");
children.push($("New Table.Child._id"));
return children;
```

Then, on save:

![image](https://user-images.githubusercontent.com/3524181/143072192-483588cf-0689-44f5-8428-12d7f67e783f.png)
